Bluetooth: controller: Send command status for unknown commands

joerchan opened the pull request:
When receiving an unknown command send the unknown command response
as a commnad status event instead of a command complete event.
A command complete event has no status field, although by convention
all command parameters has a status field as the first parameter the
command status event seems like the more fitting option.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: controller: Send command status for unknown commands

When receiving an unknown command send the unknown command response
as a command status event instead of a command complete event.
A command complete event has no status field, although by convention
all command parameters has a status field as the first parameter the
command status event seems like the more fitting option.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

carlescufi commented:
Have you checked with other implementations, such as Intel's, Qualcomm's or Broadcom's controllers to see if they handle it this way?

joerchan commented:
> Have you checked with other implementations, such as Intel's, Qualcomm's or Broadcom's controllers to see if they handle it this way?

I have checked both BT dongles that i have here. One CSR 4.0, the PTS dongle.
Both gave me a command status.

carlescufi commented:
> > Have you checked with other implementations, such as Intel's, Qualcomm's or Broadcom's controllers to see if they handle it this way?
> 
> I have checked both BT dongles that i have here. One CSR 4.0, the PTS dongle.
> Both gave me a command status.

Thanks

jhedberg commented:
I just tried with a Broadcom controller, and it also gives a Command Status

joerchan commented:
@aescolar We need https://github.com/EDTTool/EDTT/pull/17 in order to complete this PR.

joerchan committed:
west.yml: Update manifest version of EDTT for unknown command test

Update manifest to a version of EDTT that supports receiving this
response in the unknown command response test (HCI/GEV/BV-01-C).

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
